src/typescript-service.ts

```diff
@@ -143,8 +143,13 @@ export class TypeScriptService {
 		this.inMemoryFileSystem = new pm.InMemoryFileSystem(this.root);
 	}
 
-	async shutdown(params = {}, span = new Span()): Promise<void> {
+	async shutdown(params = {}, span = new Span()): Promise<null> {
 		this.projectManager.dispose();
+		return null;
+	}
+
+	async exit(params = {}, span = new Span()): Promise<void> {
```

Comment:
I would like `exit` to be handled by the part that registers the handler, because that needs to kill the process or close the socket. I would ignore `exit` in `registerLanguageHandler`